extensions/notification/queueProcessor/task.js

```diff
@@ -14,33 +22,79 @@ werelogs.configure({
     dump: config.log.dumpLevel,
 });
 
-try {
-    const destination = process.argv[2];
-    assert(destination, 'task must be started with a destination as argument');
-    // get destination auth config from environment variables
-    let destinationAuth = {
-        type: process.env.TYPE,
-        ssl: process.env.SSL === 'true',
-        protocol: process.env.PROTOCOL,
-        ca: process.env.CA,
-        client: process.env.CLIENT,
-        key: process.env.KEY,
-        keyPassword: process.env.KEY_PASSWORD,
-        keytab: process.env.KEYTAB,
-        principal: process.env.PRINCIPAL,
-        serviceName: process.env.SERVICE_NAME,
-    };
-    const isDestinationAuthEmpty = Object.values(destinationAuth)
-        .every(x => !x);
-    if (isDestinationAuthEmpty) {
-        destinationAuth = null;
+const destination = process.argv[2];
+assert(destination, 'task must be started with a destination as argument');
+// get destination auth config from environment variables
+let destinationAuth = {
+    type: process.env.TYPE,
+    ssl: process.env.SSL === 'true',
+    protocol: process.env.PROTOCOL,
+    ca: process.env.CA,
+    client: process.env.CLIENT,
+    key: process.env.KEY,
+    keyPassword: process.env.KEY_PASSWORD,
+    keytab: process.env.KEYTAB,
+    principal: process.env.PRINCIPAL,
+    serviceName: process.env.SERVICE_NAME,
+};
+const isDestinationAuthEmpty = Object.values(destinationAuth)
+    .every(x => !x);
+if (isDestinationAuthEmpty) {
+    destinationAuth = null;
+}
+const queueProcessor = new QueueProcessor(
+    mongoConfig, kafkaConfig, notifConfig, destination, destinationAuth);
+
+/**
+ * Handle ProbeServer liveness check
+ *
+ * @param {http.HTTPServerResponse} res - HTTP Response to respond with
+ * @param {Logger} log - Logger
+ * @returns {undefined}
+ */
+ function handleLiveness(res, log) {
+    if (queueProcessor.isReady()) {
+        sendSuccess(res, log);
+    } else {
+        log.error('Notification Queue Processor is not ready');
+        sendError(res, log, errors.ServiceUnavailable, 'unhealthy');
     }
-    const queueProcessor = new QueueProcessor(
-        mongoConfig, kafkaConfig, notifConfig, destination, destinationAuth);
-    queueProcessor.start();
-} catch (err) {
-    log.error('error starting notification queue processor task', {
-        method: 'notification.task.queueProcessor.start',
-        error: err,
-    });
 }
+
+async.series([
+    next => queueProcessor.start(null, next),
+    next => startProbeServer(notifConfig.probeServer, (err, probeServer) => {
+        if (err) {
+            log.error('error starting probe server', { error: err });
+            return next(err);
+        }
+        if (probeServer !== undefined) {
```

Comment:
can the opposite (probeServer == undefined) happen: it seems to me that should be an error? or is there a way to "disable" probe server from the config?

Reply:
This was a valid path due to 7.x having probe servers being opt-in env var values. In 8.x it is default on but iirc can be disabled with an env var. If it is disabled undefined is returned that is why this is here.